src/Microsoft.AspNet.Hosting.Server.Abstractions/IHttpApplication.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Hosting.Server
+{
+    /// <summary>
+    /// Represents an HttpApplication.
+    /// </summary>
+    public interface IHttpApplication<THttpContext>
```

Comment:
Isn't it a bit weird that BeginRequest returns a context as far as naming go?  But maybe something super short wouldn't lose too much meaning?

```
var context = app.Initialize(features)
try {
    await app.Process(context)
}
finally
{
   app.Dispose(context)
}